Add blog post for Home Assistant OS 10 release (#27043)
home-assistant/home-assistant.github.io: source/_posts/2023-04-18-home-assistant-os-release-10.markdown

---
layout: post
title: "Home Assistant OS 10: Better memory management and new board support"
description: "Home Assistant OS 10 released with better memory management and Hardkernel ODROID-M1 board support!"
date: 2023-04-18 00:00:00
date_formatted: "April 18, 2023"
author: Stefan Agner
comments: true
categories:
  - Announcements
og_image: /images/blog/2022-05-16-os8/social.png
---

<a href='https://github.com/home-assistant/operating-system/releases/tag/10.0'>
<img title='Home Assistant OS Release 10' alt='Home Assistant OS Release 10 Logo'
     src='/images/blog/2022-05-16-os8/social.png' style='border: 0;box-shadow: none;'>
</a>
<br><br>

**Highlights**:

- Support for Hardkernel ODROID-M1

> - Improved datadisk feature:

Author: ```suggestion
- Improved data disk feature:
```

  - Improved reliability

>   - Move from one datadisk to a new datadisk is now supported

Reviewer: ```suggestion
  - Move from one data disk to a new data disk is now supported
```
If I understand correctly, the feature is called *datadisk*, one word, but I think the actual hardware thing is still spelled in two words.

> - Improved memory management to improve overall device performance, especially under low memory situations

Reviewer: ```suggestion
- Improved memory management to improve overall device performance, especially in low memory situations
```

- Updated software packages: Linux, Docker, BlueZ, NetworkManager

For existing installations, no manual intervention is needed! You can safely
update without reading these rather technical release notes.

## Table of contents

- [Table of contents](#table-of-contents)
- [New board support: Hardkernel ODROID-M1](#new-board-support-hardkernel-odroid-m1)

> - [Improved datadisk feature](#improved-datadisk-feature)

Author: ```suggestion
- [Improved data disk feature](#improved-data-disk-feature)
```

- [Advanced memory management](#advanced-memory-management)
- [Updated software packages](#updated-software-packages)

## New board support: Hardkernel ODROID-M1

Home Assistant OS 10 supports the Hardkernel ODROID-M1 single board computer!

> Most notably the ODROID-M1 supports NVMe SSD storage natively, which makes it

Reviewer: ```suggestion
Most notably, the ODROID-M1 supports NVMe SSD storage natively, which makes it
```

another great choice to run Home Assistant on. With its quad-core CPU with up

> to 2 GHz and up to 8 GB of memory it is suitable even for demanding Home

Reviewer: ```suggestion
to 2&nbsp;GHz and up to 8&nbsp;GB of memory it is suitable even for demanding Home
```

Assistant installations.

Currently, Home Assistant can be booted off an SD-card or an eMMC. Note however
that a new boot firmware (Petitboot) is required to boot from eMMC (see
the board-specific documentation). NVMe SSD boot is currently not supported

> by the boot firmware. However, an NVMe SSD can be used through the datadisk

Author: ```suggestion
by the boot firmware. However, an NVMe SSD can be used through the data disk
```

feature.

<img src='/images/blog/2023-04-18-os10/hardkernel-odroid-m1.jpg' alt='Hardkernel ODROID-M1 single board computer with NVMe SSD plugged in'>

> ## Improved datadisk feature

Author: ```suggestion
## Improved data disk feature
```

> The datadisk feature allows extending storage by adding an external disk.

Author: ```suggestion
The data disk feature allows extending storage by adding an external disk.
```

> When using the datadisk feature, all commonly read and written data is moved

Author: ```suggestion
When using the data disk feature, all commonly read and written data is moved
```

to that storage, just Home Assistant OS itself remains on the existing storage

> (e.g. SD-card or eMMC). Home Assistant OS is a read-only operating system - the

Reviewer: ```suggestion
(e.g. SD-card or eMMC). Home Assistant OS is a read-only operating system &mdash; the
```

> (e.g. SD-card or eMMC). Home Assistant OS is a read-only operating system &mdash; the

Reviewer 2: ```suggestion
(for example, SD-card or eMMC). Home Assistant OS is a read-only operating system &mdash; the
```

ref: https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/e/eg

OS partitions are only written to when updating the operating system. This makes
sure that there is minimal wear on the existing storage.

In this OS release, together with the latest version of the Supervisor, the

> datadisk feature got more user-friendly: The datadisk choice lists the actual

Author: ```suggestion
data disk feature got more user-friendly: The data disk choice lists the actual
```

model of available data disks. In situations where previously an attached disk
would not show up, the disk will now be reliably detected and available as the

> new datadisk. Now it is also possible to move from one data disk to another:

Reviewer: ```suggestion
new data disk. Now it is also possible to move from one data disk to another:
```

Author: In the UI we use datadisk as one word. Shouldn't we stay consistent?

Reviewer: Agreed, consistency is good. But in the UI, already the second sentence uses two words for the actual hardware thing: _Moving data disks_....
It looks to me like for the feature name, consistently 1 word is used, but otherwise it should be two words (as in hard disk, floppy disk, boot disk, systems disk). Two words is correct. 
[<https://learn.microsoft.com/en-us/dotnet/api/azure.resourcemanager.compute.models.datadisk.deleteoption?view=azure-dotnet-preview>](https://learn.microsoft.com/en-us/azure/virtual-machines/managed-disks-overview)

Author: As discussed elsewhere, compound nouns should be written with a space.

> Simply connect another disk, and go to the Move datadisk dialog again. On

Reviewer: ```suggestion
Simply connect another disk, and go to the **Move datadisk** dialog again. On
```

> Simply connect another disk, and go to the **Move datadisk** dialog again. On

Author: ```suggestion
Simply connect another disk, and go to the **Move data disk** dialog again. On
```

> reboot the data will be moved, and the old data disk can be detached.

Reviewer: ```suggestion
reboot, the data will be moved and the old data disk can be detached.
```

> You can find the move datadisk feature under Settings > System > Storage in

Reviewer: ```suggestion
You can find the move datadisk feature under **Settings** > **System** > **Storage** in
```

> You can find the move datadisk feature under **Settings** > **System** > **Storage** in

Author: ```suggestion
You can find the move data disk feature under **Settings** > **System** > **Storage** in
```

the top right overflow menu.

> <img src='/images/blog/2023-04-18-os10/datadisk-new.png' alt='Screenshot showing the new datadisk selection dialog'>

Author: ```suggestion
<img src='/images/blog/2023-04-18-os10/datadisk-new.png' alt='Screenshot showing the new data disk selection dialog'>
```

## Advanced memory management

With Home Assistant OS 10 low memory, devices will generally perform better.
Firstly, we moved from zram to zswap, which allows us to use the storage as an

> actual swap space. We’ve tuned the memory management to minimize the amount of

Reviewer 2: "writes" in this sentence is a countable noun

```suggestion
actual swap space. We’ve tuned the memory management to minimize the number of
```

writes to storage (to prevent unnecessary wear on SD-cards and other flash

> memory based storage).

Reviewer 2: ```suggestion
memory-based storage).
```

We’ve also improved reliability and responsiveness in low-memory situations:
Home Assistant OS uses a new memory management mechanism named Multi-Gen LRU
along with thrashing prevention. This makes Home Assistant OS recover quickly
from a low-memory situation and remain responsive.

> In general, we recommend to use a board with at least 1 GB of memory and make

Reviewer: ```suggestion
In general, we recommend to use a board with at least 1&nbsp;GB of memory and make
```

> In general, we recommend to use a board with at least 1&nbsp;GB of memory and make

Reviewer 2: ```suggestion
In general, we recommend using a board with at least 1&nbsp;GB of memory and make
```

sure to stay well below 80% memory utilization. You can monitor the overall

> memory utilization under Settings > System > Hardware.

Reviewer: ```suggestion
memory utilization under **Settings** > **System** > **Hardware**.
```
